modules/sdk/src/test/scala/org/tessellation/sdk/infrastructure/snapshot/CurrencySnapshotCreatorSuite.scala

```diff
@@ -0,0 +1,40 @@
+package org.tessellation.sdk.infrastructure.snapshot
+
+import cats.effect.unsafe.implicits.global
+import cats.effect.{IO, Resource}
+import cats.syntax.all._
+
+import org.tessellation.ext.kryo._
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.sdk.cli.CliMethod.snapshotSizeConfig
+import org.tessellation.sdk.sdkKryoRegistrar
+import org.tessellation.security.signature.{Signed, signature}
+import org.tessellation.security.{KeyPairGenerator, SecurityProvider}
+
+import org.scalacheck.Gen
+import weaver.MutableIOSuite
+import weaver.scalacheck.Checkers
+
+object CurrencySnapshotCreatorSuite extends MutableIOSuite with Checkers {
+  type Res = (KryoSerializer[IO], SecurityProvider[IO])
+
+  def sharedResource: Resource[IO, Res] =
+    (KryoSerializer.forAsync[IO](sdkKryoRegistrar), SecurityProvider.forAsync[IO]).mapN((_, _))
+
+  def signatureProofsGen(implicit sp: SecurityProvider[IO], ks: KryoSerializer[IO]): Gen[signature.SignatureProof] = for {
```

Comment:
Do we need this? it looks like the objects created with `make` can be moved into the `for` comp then we can remove the `unsafeRunSync`

Reply:
Makes sense :+1: